packages/beacon-node/src/chain/prepareNextSlot.ts

```diff
@@ -144,7 +145,29 @@ export class PrepareNextSlotScheduler {
       if (isExecutionStateType(prepareState)) {
         const proposerIndex = prepareState.epochCtx.getBeaconProposer(prepareSlot);
         const feeRecipient = this.chain.beaconProposerCache.get(proposerIndex);
+        let updatedPrepareState = prepareState;
+        let updatedHeadRoot = headRoot;
+
         if (feeRecipient) {
+          // If we are proposing next slot, we need to predict if we can proposer-boost-reorg or not
+          const {slot: proposerHeadSlot, blockRoot: proposerHeadRoot} = this.chain.predictProposerHead(clockSlot);
```

Comment:
this would call `forkChoice.updateHead()` twice (along with the `this.chain.recomputeForkChoiceHead()` above) which is a side effect even when we don't enable `proposerBoostReorgEnabled` flag
maybe go with this one (drop `this.chain.recomputeForkChoiceHead();` above), and the result should tell if we have weak head or not

Reply:
> this would call `forkChoice.updateHead()` twice (along with the `this.chain.recomputeForkChoiceHead()` above) which is a side effect even when we don't enable `proposerBoostReorgEnabled` flag maybe go with this one (drop `this.chain.recomputeForkChoiceHead();` above), and the result should tell if we have weak head or not

`predictProposerHead()` will not call `forkChoice.updateHead()` but `forkChoice.getHead()`.
See [L188](https://github.com/ChainSafe/lodestar/blob/09d35d8e3d959aa3e2831947e0c304a175a4447a/packages/fork-choice/src/forkChoice/forkChoice.ts#L188). It's specifically written to avoid calling `updateHead()` twice